Tweak smokable volumes and add dynamic inhalation speed

BuggsWorth opened the pull request:
<!-- Please read these guidelines before opening your PR: https://docs.spacestation14.io/en/getting-started/pr-guideline -->
<!-- The text between the arrows are comments - they will not be visible on your PR. -->

## About the PR
<!-- What did you change in this PR? -->
In my last PR #30053 I had increased many of the max volumes for smokable entities to align them with the base max volume for Dan's Soaked Smokes. Instead of increasing the base inhalation rate to counter their extended duration, I have set all smokables to last a set amount of time and dynamically change their inhalation rates to match the time to finish smoking.

## Why / Balance
<!-- Why was it changed? Link any discussions or issues here. Please discuss how this would affect game balance. -->
Many chemicals and narcotics when added to cigarettes are either not abundant enough and/or do not metabolize fast enough to possess a significant threat or achieve a desired effect, allowing more room for chemicals to be added means that a higher amount of said chemicals will enter the blood stream allowing for longer or more potent effects with accumulation. This will expand upon gameplay opportunities for the entrepreneurial Chemist, Syndicate Agent with a chemical kit, or Botanist with a "Green" thumb alike. Due to these changes, the Syndicate Omnizine Smoke will be significantly buffed, making it last the set 6 minutes to smoke from the 12 minutes it used to take to completely inhale all reagents with a static inhalation amount.

## Technical details
<!-- If this is a code change, summarize at high level how your new code works. This makes it easier to review. -->
The Smoking System now calculates the inhalation amount instead of using the Smokable Component's fixed inhalation amount, it will change depending on the volume of chemicals contained within the smokable making sure that you inhale the correct amount to finish all reagents within the set time frame of 360 seconds as defined by SmokableDuration float. In practice a full cigarette with 40u of chemicals will inhale ~0.33u of the total volume per update/tick in the Smoking System.

In addition to the dynamic inhalation amount, adjusted the max volume of the base cigarette/cigar entity to match the volume of "Dan's Soaked Smokes" (40u max), tweaked the max volume of other smokables such as blunts and joints to make more room for any additional chemicals added from plant mutations, and doubled the amount of extra chemicals to "Dan's Soaked Smoke's" smokes.

## Media
<!-- 
PRs which make ingame changes (adding clothing, items, new features, etc) are required to have media attached that showcase the changes.
Small fixes/refactors are exempt.
Any media may be used in SS14 progress reports, with clear credit given.

If you're unsure whether your PR will require media, ask a maintainer.

Check the box below to confirm that you have in fact seen this (put an X in the brackets, like [X]):
-->

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
<!--
List any breaking changes, including namespace, public class/method/field changes, prototype renames; and provide instructions for fixing them. This will be pasted in #codebase-changes.
-->

**Changelog**
<!--
Make players aware of new features and changes that could affect how they play the game by adding a Changelog entry. Please read the Changelog guidelines located at: https://docs.spacestation14.io/en/getting-started/pr-guideline#changelog
-->

<!--
Make sure to take this Changelog template out of the comment block in order for it to show up.
:cl:
- add: Added fun!
- remove: Removed fun!
- tweak: Changed fun!
- fix: Fixed fun!
-->
:cl:
- tweak: Smokables now all last a set 6 minutes allowing you to inhale more at once and can hold an additional 20u of chemicals
- tweak: Doubled the chemicals in all of "Dan's Soaked Smoke's"

Boaz1111 commented:
you should rewrite the second part of the changelog to not be RP, RP changelogs are usually frowned upon because they’re not direct and to the point enough

BuggsWorth commented:
> you should rewrite the second part of the changelog to not be RP, RP changelogs are usually frowned upon because they’re not direct and to the point enough

Done!

Golinth commented:
I like this change a lot. Syndie smokes are very rarely used for the health, and the only time I see them is a new player or someone trying to farm omni

deepdarkdepths commented:
I love syndie smokes, and despite how bad they are, used them often. This would be a welcome change.

dbrezack commented:
Would it make sense to increase the volume of cigars and base the timer off of total volume ? 

That way cigars would take longer to smoke in game than a cigarette instead of 6 minutes flat for everything

BuggsWorth commented:
> Would it make sense to increase the volume of cigars and base the timer off of total volume ?

This is how it used to function before I added the timer, inhalation rate was fixed and made it so the higher the volume the longer the smoke. I added the dynamic inhalation rate to make it so if anything was added by another player or through plant mutations to the smoke, you would ingest a greater amount of chemicals per inhalation.

> That way cigars would take longer to smoke in game than a cigarette instead of 6 minutes flat for everything

Cigarettes and Cigars all use the same "BaseCigar" parent entity meaning they had the same volume and therefor duration (with exception to soaked smokes), IIRC Cigarettes and Cigars both lasted 4 minutes before this change making this an increase in time to smoke either way.

Raccoononi commented:
I would love for this change to be in game, I am in a mad chemist rabbit hole right now and would absolutely love to explore the potential of microdosing chems through cigarettes.

The use cases I'm already thinking of: Tranexamic acid would stop bleeding as someone is in combat, dexaline plus would counter airloss steadily, and, well, meth would provide a constant speed boost.

This would open many opportunities, and I want to comment on this PR again to bring attention to it.

metalgearsloth commented on Content.Server/Nutrition/EntitySystems/SmokingSystem.cs:
```diff
@@ -35,13 +35,15 @@ public sealed partial class SmokingSystem : EntitySystem
         /// <summary>
         ///     We keep a list of active smokables, because iterating all existing smokables would be dumb.
         /// </summary>
         private readonly HashSet<EntityUid> _active = new();
+        private readonly Dictionary<EntityUid, FixedPoint2> _initialVolume = new();
```
This isn't going to serialize properly so you can't use this.

BuggsWorth replied:
> This isn't going to serialize properly so you can't use this.

I can change this to work a different way, can you tell me why it wont serialize properly? Not very C# literate and it never threw any errors in testing that I could see, this would be my first contribution to anything as well.

BuggsWorth commented:
Dont have time to work on this atm so ill close the PR and re-open/re-make the PR when ive fixed the issues
